components/renku_data_services/project/db.py

```diff
@@ -142,15 +142,27 @@ async def insert_project(
         """Insert a new project entry."""
         if not session:
             raise errors.ProgrammingError(message="A database session is required")
-        ns = await self.group_repo.get_namespace_by_slug(user, project.namespace)
+        ns = await session.scalar(
+            select(schemas.NamespaceORM).where(schemas.NamespaceORM.slug == project.namespace.lower())
+        )
         if not ns:
             raise errors.MissingResourceError(
                 message=f"The project cannot be created because the namespace {project.namespace} does not exist"
             )
+        if not ns.group_id and not ns.user_id:
+            raise errors.ProgrammingError(message="Found a namespace that has no group or user associated with it.")
 
         if user.id is None:
             raise errors.Unauthorized(message="You do not have the required permissions for this operation.")
 
+        resource_type = ResourceType.group if ns.group and ns.group_id else ResourceType.user_namespace
+        resource_id = ns.group_id or ns.user_id or ""
```

Comment:
TODO: update `resource_id` to be the `ns.id` when `ResourceType` is equal to `user`.